Estonian-6.17.0-1

kristoheero opened the pull request:
* XTE-355 / Backlog #152: Security Server: Improved message exchange performance at a time when periodical timestamping is performed
* XTE-368: Added new security server metapackage xroad-securityserver-ee with default configuration for Estonian instances
* XTE-375: Security Server / Central Server: Enabled HttpOnly flag and set security flag to true for the session cookies
* XTE-376: Security Server: Fixed system resource leak of the monitoring component
* XTE-380: Security Server: Fixed audit logging of the restore process

Modified documents:
* IG-CS
* IG-SS

Commit:
XTE-368: Added new security server metapackage xroad-securityserver-ee with default configuration for Estonian instances

Commit:
XTE-376: Security Server: Fixed system resource leak of the monitoring component

Commit:
XTE-375: Security Server / Central Server: Enabled HttpOnly flag and set security flag to true for the session cookies

Commit:
XTE-355 / Backlog #152: Security Server: Improved message exchange performance at a time when periodical timestamping is performed.
Also, fixed messagelog junit tests and added logback configuration.

Commit:
XTE-380: Fixed audit logging of the restore process.
Fixed one NPE.

Commit:
Changelog for estonian-6.17.0-1

jhyoty commented on src/addons/messagelog/src/main/java/ee/ria/xroad/proxy/messagelog/LogManager.java:
```diff
@@ -212,62 +202,81 @@ protected Props getTimestamperImpl() {
 
+            try {
+                if (message instanceof SetTimestampingStatusMessage) {
+                    logManager.setTimestampingStatus((SetTimestampingStatusMessage) message);
```
An actor should not directly call a method of an another actor (it should send a message). In addition, this is a potential concurrency issue, since LogManager's internal state is modified.

kristoheero replied:
For the sake of clarity and comprehensibility the communication between LogManager and TaskQueue is completely refactored now and also some additional bugs (related with timestamp-immediately mode) are fixed. It seems, that initial performance issue has been arised already from pull request finish-6.7.10, where previous solution was rebuild and timestamping post processing (modification of DB records) was moved from TaskQueue actor to the LogManager actor.

jhyoty commented on src/addons/messagelog/src/main/java/ee/ria/xroad/proxy/messagelog/LogManager.java:
```diff
@@ -212,62 +202,81 @@ protected Props getTimestamperImpl() {
+                if (message instanceof SetTimestampingStatusMessage) {
+                    logManager.setTimestampingStatus((SetTimestampingStatusMessage) message);
+                } else if (message instanceof SaveTimestampedDataMessage) {
+                    logManager.saveTimestampRecord(((SaveTimestampedDataMessage) message).getTimestampSucceeded());
```
An actor should not directly call a method of an another actor.

kristoheero replied:
See previous comment.

Commit:
XTE-355 / Backlog #152: Security Server: Refactored communication between
LogManager and TaskQueue. In 'timestamp immediately' mode do not take into account
acceptable timestamp failure period. Codestyle.

iluwatar commented on src/addons/messagelog/src/main/java/ee/ria/xroad/proxy/messagelog/LogRecordManager.java:
```diff
@@ -43,212 +48,235 @@
+        log.trace("setMessageRecordsTimestamped({})", messageRecords.length);
+
+        int storedCount = 0;
+        PreparedStatement stmt = connection.prepareStatement("UPDATE logrecord SET timestamprecord = ?, "
```
This line is raising a blocker level bug in Sonarqube: `Use try-with-resources or close this "PreparedStatement" in a "finally" clause.`

kristoheero replied:
True. I committed this fix into the branch estonian-6.17.0-1 and made new pull request Estonian-6.17.0-3.
